overlord/fdestate/backend/reseal.go

```diff
@@ -114,49 +127,87 @@ func ResealKeyForBootChains(method device.SealingMethod, rootdir string, params
 	return nil
 }
 
-func resealRunObjectKeys(pbc boot.PredictableBootChains, authKeyFile string, roleToBlName map[bootloader.Role]string) error {
+func resealRunObjectKeys(updateState stateUpdater, pbc boot.PredictableBootChains, authKeyFile string, roleToBlName map[bootloader.Role]string) error {
 	// get model parameters from bootchains
 	modelParams, err := boot.SealKeyModelParams(pbc, roleToBlName)
 	if err != nil {
 		return fmt.Errorf("cannot prepare for key resealing: %v", err)
 	}
 
+	numModels := len(modelParams)
+	if numModels < 1 {
+		return fmt.Errorf("at least one set of model-specific parameters is required")
+	}
+
+	pcrProfile, err := secbootBuildPCRProtectionProfile(modelParams)
+	if err != nil {
+		return err
+	}
+
+	var models []secboot.ModelForSealing
+	for _, m := range modelParams {
+		models = append(models, m.Model)
+	}
+
 	// list all the key files to reseal
 	keyFiles := []string{device.DataSealedKeyUnder(boot.InitramfsBootEncryptionKeyDir)}
 
 	resealKeyParams := &secboot.ResealKeysParams{
-		ModelParams:          modelParams,
+		PCRProfile:           pcrProfile,
 		KeyFiles:             keyFiles,
 		TPMPolicyAuthKeyFile: authKeyFile,
 	}
 	if err := secbootResealKeys(resealKeyParams); err != nil {
 		return fmt.Errorf("cannot reseal the encryption key: %v", err)
 	}
 
+	if err := updateState("run+recover", "all", []string{"run", "recover"}, models, pcrProfile); err != nil {
```

Comment:
perhaps same thing about bare `"all"`, what if we simply have type for it?

Reply:
Let's leave a TODO about maybe. We'll try to figure it out later when we have a better feeling of what's needed.

Reply:
Done